app/models/manageiq/providers/inventory/persister/builder/persister_helper.rb

```diff
@@ -119,4 +131,14 @@ def references(collection)
   def name_references(collection)
     target.try(:manager_refs_by_association).try(:[], collection).try(:[], :name).try(:to_a) || []
   end
+
+  private
+
+  def add_collection_for_manager(manager_type, collection_name, extra_properties = {}, settings = {}, &block)
+    parent_manager = send("#{manager_type}_manager")
+    settings.reverse_merge!(:shared_properties => {:parent => parent_manager})
```

Comment:
I don't understand why the complexity is needed around properties, settings, adv_settings, shared_properties, etc... when they are all just combined together into one flat hash at the end.

Might be too much change for this PR but I think this can be a lot simpler than it currently is

Reply:
yeah, smaller change PRs are easier to create and review.